File: javascript/easy/validPalindromeII.js
/**
 * @param {string} s
 * @return {boolean}
 */
var validPalindrome = function(s) {
    return isPalindrome(s, 0, s.length-1, 0);
};

var isPalindrome = function(s, i, j, count) {
    if(i >= j) return true;
    if(s[i] === s[j]) {
        return isPalindrome(s, i+1, j-1, count);
    } else {
        return (count < 1) && (isPalindrome(s, i+1, j, count+1) || isPalindrome(s, i, j-1, count+1));
    }
};

File: ruby/easy/valid_palindrome_ii.rb
# @param {String} s
# @return {Boolean}
def valid_palindrome(s)
    return is_palindrome(s, 0, s.length-1, 0)
end

def is_palindrome(s, i, j, count)
    return true if (i >= j)
    if(s[i] == s[j])
        return is_palindrome(s, i+1, j-1, count)
    else
        return (count < 1) && (is_palindrome(s, i+1, j, count+1) || is_palindrome(s, i, j-1, count+1))
    end
end
